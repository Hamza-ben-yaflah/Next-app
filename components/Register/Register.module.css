.Register {
  height: 100vh;
  justify-content: center;
  align-items: center;
  display: flex;
}
.Register_form_button {
  width: 100%;
}
