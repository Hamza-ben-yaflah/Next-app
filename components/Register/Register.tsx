import { Button, Form, Input, Select } from "antd";
import React from "react";
import styles from "./Register.module.css";
import Link from "next/link";

const { Option } = Select;
function Register() {
  const prefixSelector = (
    <Form.Item name="prefix" noStyle>
      <Select style={{ width: 80 }}>
        <Option value="216">+216</Option>
      </Select>
    </Form.Item>
  );
  return (
    <div className={styles.Register}>
      <Form
        name="register"
        initialValues={{
          prefix: "216",
        }}
        scrollToFirstError
      >
        <Form.Item
          name="email"
          label="E-mail"
          rules={[
            {
              type: "email",
              message: "The input is not valid E-mail!",
            },
            {
              required: true,
              message: "Please input your E-mail!",
            },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          name="password"
          label="Password"
          rules={[
            {
              required: true,
              message: "Please input your password!",
            },
          ]}
          hasFeedback
        >
          <Input.Password />
        </Form.Item>

        <Form.Item
          name="confirm"
          label="Confirm Password"
          dependencies={["password"]}
          hasFeedback
          rules={[
            {
              required: true,
              message: "Please confirm your password!",
            },
            ({ getFieldValue }) => ({
              validator(_, value) {
                if (!value || getFieldValue("password") === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  new Error("The two passwords that you entered do not match!")
                );
              },
            }),
          ]}
        >
          <Input.Password />
        </Form.Item>
        <Form.Item
          name="username"
          label="Username"
          rules={[
            {
              required: true,
              message: "Please input your name!",
              whitespace: true,
            },
          ]}
        >
          <Input />
        </Form.Item>
        <Form.Item
          name="phone"
          label="Phone Number"
          rules={[
            { required: true, message: "Please input your phone number!" },
          ]}
        >
          <Input addonBefore={prefixSelector} style={{ width: "100%" }} />
        </Form.Item>
        <Form.Item
          name="gender"
          label="Gender"
          rules={[{ required: true, message: "Please select gender!" }]}
        >
          <Select placeholder="select your gender">
            <Option value="male">Male</Option>
            <Option value="female">Female</Option>
            <Option value="other">Other</Option>
          </Select>
        </Form.Item>
        <Form.Item>
          <Link href="/ChoicePage" passHref>
            <Button
              type="primary"
              htmlType="submit"
              className={styles.Register_form_button}
            >
              <a>Register</a>
            </Button>
          </Link>
        </Form.Item>
      </Form>
    </div>
  );
}

export default Register;
