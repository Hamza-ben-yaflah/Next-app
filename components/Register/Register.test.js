import { render, screen } from "@testing-library/react";
import Register from "./Register";
import renderer from "react-test-renderer";
test("render the Register Component", () => {
  render(<Register />);
  const Reg = screen.getByTestId("register");
  expect(Reg).toBeInTheDocument();
});

test("render the Register Ui", () => {
  const Reg = renderer.create(<Register />).toJSON();
  expect(Reg).toMatchSnapshot();
});
