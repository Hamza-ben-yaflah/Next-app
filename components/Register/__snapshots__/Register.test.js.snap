// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Register Ui 1`] = `
<div
  className="Register"
  data-testid="register"
>
  <form
    className="ant-form ant-form-horizontal"
    id="register"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_email"
          title="E-mail"
        >
          E-mail
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="register_email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_password"
          title="Password"
        >
          Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input-password"
              onMouseUp={[Function]}
            >
              <input
                action="click"
                className="ant-input"
                id="register_password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <span
                  aria-label="eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="img"
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_confirm"
          title="Confirm Password"
        >
          Confirm Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input-password"
              onMouseUp={[Function]}
            >
              <input
                action="click"
                className="ant-input"
                id="register_confirm"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <span
                  aria-label="eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="img"
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_username"
          title="Username"
        >
          Username
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="register_username"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_phone"
          title="Phone Number"
        >
          Phone Number
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-group-wrapper"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <span
                className="ant-input-wrapper ant-input-group"
              >
                <span
                  className="ant-input-group-addon"
                >
                  <div
                    className="ant-select ant-select-single ant-select-show-arrow"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "width": 80,
                      }
                    }
                  >
                    <div
                      className="ant-select-selector"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <span
                        className="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="register_prefix_list_0"
                          aria-autocomplete="list"
                          aria-controls="register_prefix_list"
                          aria-haspopup="listbox"
                          aria-owns="register_prefix_list"
                          autoComplete="off"
                          className="ant-select-selection-search-input"
                          id="register_prefix"
                          onChange={[Function]}
                          onCompositionEnd={[Function]}
                          onCompositionStart={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onPaste={[Function]}
                          readOnly={true}
                          role="combobox"
                          style={
                            Object {
                              "opacity": 0,
                            }
                          }
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        className="ant-select-selection-item"
                        title="+216"
                      >
                        +216
                      </span>
                    </div>
                    <span
                      aria-hidden={true}
                      className="ant-select-arrow"
                      onMouseDown={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        className="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </span>
                <input
                  className="ant-input"
                  id="register_phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="text"
                  value=""
                />
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_gender"
          title="Gender"
        >
          Gender
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-select ant-select-single ant-select-show-arrow"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
            >
              <div
                className="ant-select-selector"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <span
                  className="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="register_gender_list_0"
                    aria-autocomplete="list"
                    aria-controls="register_gender_list"
                    aria-haspopup="listbox"
                    aria-owns="register_gender_list"
                    autoComplete="off"
                    className="ant-select-selection-search-input"
                    id="register_gender"
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onPaste={[Function]}
                    readOnly={true}
                    role="combobox"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  className="ant-select-selection-placeholder"
                >
                  select your gender
                </span>
              </div>
              <span
                aria-hidden={true}
                className="ant-select-arrow"
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <a
              className="ant-btn ant-btn-primary Register_form_button"
              href="/ChoicePage"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <span>
                Register
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
